WooCommerce/Classes/Tools/Logging/WooCrashLoggingStack.swift

```diff
@@ -0,0 +1,145 @@
+import Foundation
+import AutomatticTracks
+import Storage
+import Yosemite
+
+/// A wrapper around the logging stack – provides shared initialization and configuration for Tracks Crash and Event Logging
+struct WooCrashLoggingStack: CrashLoggingStack {
+    static let QueuedLogsDidChangeNotification = NSNotification.Name("WPCrashLoggingQueueDidChange")
+
+    let crashLogging: AutomatticTracks.CrashLogging
+    let eventLogging: EventLogging
+
+    private let crashLoggingDataProvider = WCCrashLoggingDataProvider()
+    private let eventLoggingDataProvider = WCEventLoggingDataSource()
+    private let eventLoggingDelegate = WCEventLoggingDelegate()
+
+    init() {
+        let eventLogging = EventLogging(dataSource: eventLoggingDataProvider, delegate: eventLoggingDelegate)
+
+        self.eventLogging = eventLogging
+        self.crashLogging = AutomatticTracks.CrashLogging(dataProvider: crashLoggingDataProvider, eventLogging: eventLogging)
+
+        /// Upload any remaining files any time the app becomes active
+        let willEnterForeground = UIApplication.willEnterForegroundNotification
+        NotificationCenter.default.addObserver(forName: willEnterForeground, object: nil, queue: nil, using: self.willEnterForeground)
+        _ = try? crashLogging.start()
+    }
+
+    func crash() {
+        crashLogging.crash()
+    }
+
+    func logMessage(_ message: String, properties: [String: Any]?, level: SeverityLevel) {
+        crashLogging.logMessage(message, properties: properties, level: sentrySeverity(with: level))
+    }
+
+    func logError(_ error: Error) {
+        crashLogging.logError(error)
+    }
+
+    func logError(_ error: Error, userInfo: [String: Any]? = nil, level: SeverityLevel = .error) {
+        crashLogging.logError(error, userInfo: userInfo, level: sentrySeverity(with: level))
+    }
+
+    func logFatalErrorAndExit(_ error: Error, userInfo: [String: Any]? = nil, level: SeverityLevel = .error) -> Never {
```

Comment:
Feels a bit odd that the name of this method contains `fatalError` but also accepts a `SeverityLevel`. Which means that we can for example call `logFatalErrorAndExit(…, userInfo: nil, level: .info)` too, which I'm not sure makes sense (and even if it made sense in some scenario, it would feels a bit confusing 🤔 )

Reply:
Agreed, I added the fatal to this method towards the end and didn't realize the level was still there.